test/rego_checks_test.go

```diff
@@ -0,0 +1,744 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	checks "github.com/aquasecurity/trivy-checks"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/accessanalyzer"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/athena"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/cloudtrail"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/codebuild"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/config"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/documentdb"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/dynamodb"
+	"github.com/aquasecurity/trivy/pkg/iac/providers/aws/s3"
+	"github.com/aquasecurity/trivy/pkg/iac/rego"
+	"github.com/aquasecurity/trivy/pkg/iac/rules"
+	"github.com/aquasecurity/trivy/pkg/iac/scan"
+	"github.com/aquasecurity/trivy/pkg/iac/state"
+	trivyTypes "github.com/aquasecurity/trivy/pkg/iac/types"
+	ruleTypes "github.com/aquasecurity/trivy/pkg/iac/types/rules"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func scanState(t *testing.T, regoScanner *rego.Scanner, s state.State, checkID string, expected bool) {
+	results, err := regoScanner.ScanInput(context.TODO(), rego.Input{
+		Contents: s.ToRego(),
+	})
+	require.NoError(t, err)
+
+	var found bool
+	for _, result := range results {
+		if result.Status() == scan.StatusFailed && result.Rule().AVDID == checkID {
+			found = true
+		}
+	}
+
+	if expected {
+		assert.True(t, found, "Rule should have been found")
+	} else {
+		assert.False(t, found, "Rule should not have been found")
+	}
+}
+
+func TestAWSRegoChecks(t *testing.T) {
+	type testCase struct {
+		name     string
+		input    state.State
+		expected bool
+	}
+
+	tests := map[string][]testCase{
+		"AVD-AWS-0175": {
+			// TODO: Trivy does not export empty structures into Rego
+			// {
+
+			// 	name:     "No analyzers enabled",
+			// 	input:    state.State{AWS: aws.AWS{AccessAnalyzer: accessanalyzer.AccessAnalyzer{}}},
+			// 	expected: true,
+			// },
```

Comment:
Pardon my memory but did we ever decide how to proceed on this?

Reply:
We've discussed this in DM, but we haven't come to a conclusion yet.